packages/core/src/core/client.ts

```diff
@@ -123,6 +124,10 @@ export class GeminiClient {
     return this.chat;
   }
 
+  commitCancelledResponse(finalPartialContent: HistoryContent) {
```

Comment:
My concern here is with how we are doing implicit caching. I'm not sure how a user cancelled or errored out stream is handled @cornmander